src/agni/client/view/ChatView.java

```diff
@@ -1,36 +1,166 @@
 package agni.client.view;
 
+import agni.client.AgniClient;
+import agni.client.HeartbeatMonitor;
 import agni.client.action.*;
 import agni.server.sender.StatusSender.Status;
-
-public class ChatView implements AgniClientView {
-
+import charva.awt.BorderLayout;
+import charva.awt.Container;
+import charva.awt.event.ActionEvent;
+import charva.awt.event.ActionListener;
+import charva.awt.event.KeyEvent;
+import charva.awt.event.KeyListener;
+import charvax.swing.JFrame;
+import charvax.swing.JLabel;
+import charvax.swing.JMenu;
+import charvax.swing.JMenuBar;
+import charvax.swing.JMenuItem;
+import charvax.swing.JTextArea;
+import charvax.swing.JTextField;
+
+public class ChatView extends JFrame implements AgniClientView, ActionListener, KeyListener {
+
+    private AgniClient client;
     private InfoRequestActionHandler infoRequestActionHandler;
     private UserActionHandler userActionHandler;
     private ChatActionHandler chatActionHandler;
     private FileActionHandler fileActionHandler;
-    private HeartbeatActionHandler heartbeatActionHandler;
-
-    public ChatView(InfoRequestActionHandler infoRequestActionHandler,
+    private HeartbeatMonitor serverMonitor;
+    private JTextField inputLine;
+    private JTextArea outputArea;
+    private final int WIDTH = 80;
+    private final int HEIGHT = 24;
```

Comment:
Is there a way to make this dynamic? It doesn't really matter but it would be nice to have that